modules/custom/az_migration/src/Plugin/migrate/source/AZParagraphsItem.php

```diff
@@ -29,8 +43,11 @@ public function query() {
         ])
       ->fields('pr', ['revision_id']);
     $query->innerJoin('paragraphs_item_revision', 'pr', static::JOIN);
-    // Omit archived (deleted) paragraphs.
-    $query->condition('p.archived', 0);
+
+    if (empty($this->allowArchivedParagraphs)) {
```

Comment:
I think this is wrong.  `empty(FALSE)` would return `TRUE`.
```suggestion
    if (!empty($this->allowArchivedParagraphs)) {
```

Reply:
Nevermind.  I think this is OK.  `empty()` is just confusing here.